extensions/amp-access/0.1/amp-access.js

```diff
@@ -542,38 +342,25 @@ export class AccessService {
     if (!this.enabled_) {
       return null;
     }
-    return this.lastAuthorizationPromise_.then(() => {
-      if (!this.authResponse_) {
+    // remove catch?
+    return this.lastAuthorizationPromises_.then(() => {
```

Comment:
It seems like this could yield `null` too fast, since it only waits for the fastest promise, if my reading of this is right:
```
this.lastAuthorizationPromises_ = this.sources_.length > 1 ?
      Promise.race(promises) : Promise.all(promises);
```

One option is we can have the `all` promise for them all. Or another option would be to first parse the first component of the `field` and wait for that specific promise.

Reply:
Yeah, good point, I was wondering about this. I think we basically have three options.

1. As now, yield null. My reading of the call sites suggests that we may send some initial requests with null values but that once the value is correct they will have the correct values, but maybe that's not sufficient.

2. Use an all promise. One reason I moved away from all() is that if any one request fails as currently built the whole thing never completes, so that could mean one bad request would cause _all_ authdata fields to fail. We could change that behavior, but it started to break other things when I tried this.

3. Parsing the namespace. I think I can do this easily enough but wasn't sure if there were cases where authdata fields might be expressions (e.g. foo OR bar). Is the possible? Otherwise, this might be our best option and should be okay, maybe just a little clunky.

Reply:
I think a more direct protection for "one request failed" is a good idea, rather than relying on implicit semantics of race.

The access data is always a simple `object.field` access.

However, this brings up a bigger question. What do we do with templates? Do we wait for all requests to render them? I don't think we can currently re-render...

Reply:
So if I understand correctly, are you a fan of option 3, or were you thinking option 2 but changing the current behavior where a failed access call just never resolves?

Just to be clear, option 3 should be easy-ish to apply to getAuthdataField but may still lead to templates and sections that initially see null and then see a valid value. (My reading is that templates should re-render as each response comes in.)